protocol_common.go

```diff
@@ -355,6 +356,25 @@ func (h *Headscale) handleAuthKeyCommon(
 
 			return
 		}
+
+		aclTags := pak.toProto().AclTags
+		if len(aclTags) > 0 {
+			// This conditional preserves the existing behaviour, although SaaS would reset the tags on auth-key login
+			err = h.SetTags(machine, aclTags)
+		}
+
+		if err != nil {
```

Comment:
This error check should be inside the `if len(aclTags)` closure so we don't risk any weird scoping behaviour.